runtime/js/30_fs.js

```diff
@@ -148,48 +154,101 @@
     path,
     options = {},
   ) {
-    ops.op_remove_sync({
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    ops.op_remove_sync(
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   async function remove(
     path,
     options = {},
   ) {
-    await core.opAsync("op_remove_async", {
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    await core.opAsync(
+      "op_remove_async",
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   function renameSync(oldpath, newpath) {
-    ops.op_rename_sync({
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    ops.op_rename_sync(
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
   async function rename(oldpath, newpath) {
-    await core.opAsync("op_rename_async", {
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    await core.opAsync(
+      "op_rename_async",
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
+  function createByteStruct(types) {
+    // types can be "date", "bool" or "u64".
+    // `?` prefix means optional on windows.
+    let offset = 0;
+    let str =
+      'const unix = Deno.build.os === "darwin" || Deno.build.os === "linux"; return {';
+    for (let [name, type] of ObjectEntries(types)) {
+      const optional = type.startsWith("?");
+      if (optional) type = type.slice(1);
+
+      if (type == "u64") {
+        if (!optional) {
+          str += `${name}: view[${offset}] + view[${offset + 1}] * 2**32,`;
+        } else {
+          str += `${name}: (unix ? (view[${offset}] + view[${
+            offset + 1
+          }] * 2**32) : (view[${offset}] + view[${
+            offset + 1
+          }] * 2**32) || null),`;
```

Comment:
Are both branches of this ternary expression the same?

Reply:
nope, on windows we want `0` value to be `null`

Reply:
Got it, had 't checked the operator precedence of that.